### PR TITLE
Cleanup platform-specific ifdefs

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

**Comment by jkotas** [comment]
Inspired by LoongArch64 PR reviews

File: src/coreclr/vm/exceptionhandling.cpp [hunk 1]
```diff
@@ -3276,10 +3276,8 @@ DWORD_PTR ExceptionTracker::CallHandler(
     StackFrame             sf,
     EE_ILEXCEPTION_CLAUSE* pEHClause,
     MethodDesc*            pMD,
-    EHFuncletType funcletType
-    X86_ARG(PCONTEXT pContextRecord)
-    ARM_ARG(PCONTEXT pContextRecord)
-    ARM64_ARG(PCONTEXT pContextRecord)
+    EHFuncletType          funcletType,
+    PCONTEXT               pContextRecord)
     )
```
Comment (jkotas) [inline src/coreclr/vm/exceptionhandling.cpp:3281]:
```suggestion
```

**Comment by shushanhf** [comment]
Hi @jkotas 
You had cleaned the redundant defination liking `IMAGE_FILE_XXX`.

How about the relocation liking `IMAGE_REL_ARM64_PAGEOFFSET_12A` within the `src/coreclr/inc/clrnt.h` and `src/coreclr/pal/inc/rt/ntimage.h` ?

**Comment by jkotas** [comment]
> How about the relocation liking IMAGE_REL_ARM64_PAGEOFFSET_12A within the src/coreclr/inc/clrnt.h

I have pushed a commit to cleanup these as well.

**Review by BruceForstall — APPROVED** [review]
LGTM

**Comment by jkotas** [comment]
I do not think that it is related.

We require VS 2022 for build https://github.com/dotnet/runtime/blob/main/docs/workflow/requirements/windows-requirements.md#visual-studio . The log that you have shared suggests that the CI machine is using VS 20219. I guess the CI machine is misconfigured. What is the job that is hitting this break?

**Comment by AndyAyersMS** [comment]
https://github.com/dotnet/runtime/pull/65675/checks?check_run_id=5307915150

**Comment by AndyAyersMS** [comment]
Seems like this comes from `eng\pipelines\common\xplat-setup.yml`?

**Pull request merged**